test/core/blockchain/weight_calculator/weight_calculator_test.cpp

```diff
@@ -0,0 +1,93 @@
+
+using fc::blockchain::weight::WeightCalculatorImpl;
+using fc::primitives::address::Address;
+using fc::primitives::block::BlockHeader;
+using fc::primitives::tipset::Tipset;
+using fc::storage::ipfs::InMemoryDatastore;
+using fc::vm::actor::Actor;
+using fc::vm::actor::ActorSubstateCID;
+using fc::vm::actor::kStoragePowerAddress;
+using fc::vm::actor::kStoragePowerCodeCid;
+using fc::vm::actor::builtin::storage_power::StoragePowerActorState;
+using fc::vm::state::StateTreeImpl;
+using Weight = fc::primitives::BigInt;
+using fc::power::Power;
+
+struct Params {
+  Weight parent_weight;
+  Power network_power;
+  size_t block_count;
+  Weight expected_weight;
+};
+
+struct WeightCalculatorTest : ::testing::TestWithParam<Params> {};
+
+TEST_P(WeightCalculatorTest, GoCompatibility) {
+  auto &params = GetParam();
+  auto ipld = std::make_shared<InMemoryDatastore>();
+  auto some_cid = "010001020001"_cid;
+  EXPECT_OUTCOME_TRUE(state_cid,
+                      ipld->setCbor(StoragePowerActorState{
+                          .total_network_power = params.network_power,
+                          .miner_count = {},
+                          .escrow_table_cid = some_cid,
+                          .cron_event_queue_cid = some_cid,
+                          .po_st_detected_fault_miners_cid = some_cid,
+                          .claims_cid = some_cid,
+                          .num_miners_meeting_min_power = {},
+                      }));
+  StateTreeImpl state_tree{ipld};
+  EXPECT_OUTCOME_TRUE_1(state_tree.set(kStoragePowerAddress,
+                                       Actor{
+                                           .code = kStoragePowerCodeCid,
+                                           .head = ActorSubstateCID{state_cid},
+                                           .nonce = {},
+                                           .balance = {},
+                                       }));
+  EXPECT_OUTCOME_TRUE(state_root, state_tree.flush());
+  Tipset tipset{{},
+                {params.block_count,
+                 BlockHeader{
+                     .miner = Address::makeFromId(0),
+                     .ticket = {},
+                     .epost_proof = {},
+                     .parents = {},
+                     .parent_weight = params.parent_weight,
+                     .height = {},
+                     .parent_state_root = state_root,
+                     .parent_message_receipts = some_cid,
+                     .messages = some_cid,
+                     .bls_aggregate = {},
+                     .timestamp = {},
+                     .block_sig = {},
+                     .fork_signaling = {},
+                 }},
+                {}};
+
+  EXPECT_OUTCOME_EQ(WeightCalculatorImpl{ipld}.calculateWeight(tipset),
+                    params.expected_weight);
+}
+
+INSTANTIATE_TEST_CASE_P(WeightCalculatorTestCases,
+                        WeightCalculatorTest,
+                        ::testing::Values(Params{100, 200, 1, 2071},
```

Comment:
Test coverage reports about low coverage. Can you add zero network_power test, please.